### ISSUE TITLE
Rename transaction retry related system variables

### ISSUE THREAD

**Issue by djshow832** [pingcap/tidb#23375]
## Background

As mentioned in https://github.com/pingcap/docs-cn/pull/5725, there are 3 variables and configurations to control the retrial of transactions:

- Optimistic transactions: `tidb_disable_txn_auto_retry` and `tidb_retry_limit`.
- Pessimistic transactions: `max-retry-count`.

Although the document is polished, users may be still confused if they don't refer to the document.

## Proposal

I propose to rename these variables or configurations to make them clearer.

- Optimistic transactions: `tidb_enable_opt_txn_retry` and `tidb_opti_txn_retry_limit`.
- Pessimistic transactions: `tidb_pess_txn_retry_limit`.

To be compatible with older versions, current variables should be marked as deprecated, instead of removed directly. The old names won't be shown in `show variables` or `show config` statements.

**Comment by djshow832** [comment]
@SunRunAway @cfzjywxk  What do you think? Any better ideas?

**Comment by morgo** [comment]
Why not use `tidb_txn_retry_limit` for both pessimistic and optimistic? I don't think it is good to abbreviate to `opti` or `pess`, since it is not like `txn` where the abbreviation is well established.

**Comment by djshow832** [comment]
I agree with you on that the abbreviation is not suitable. I'm concerned by whether the retry limit of optimistic and pessimistic transactions can be always the same. What do you think by your experience? @cfzjywxk

**Comment by djshow832** [comment]
Besides, can we use `tidb_enable_txn_retry` for both pessimistic and optimistic?

**Comment by morgo** [comment]
> Besides, can we use `tidb_enable_txn_retry` for both pessimistic and optimistic?

I think it's safer to use the same name for both (even if the implementation logic ends up being different). If you use different names there is a higher risk of misconfiguration.

If users want different behaviors for their pessimistic vs. optimistic transactions, they will need to use session variables.

**Comment by cfzjywxk** [comment]
The retry is a a bit different as the pessimistic retry is statement level retry and the optimistic retry is transaction level, the optimistic retry may be deprecated in the future.  Maybe we could leave just one "retry" variable, I'm not sure about the proper name currently,  or just keep the `tidb_retry_limit` name?

Besides, we may need to consider about the incompatibility issues for users using the original names or configs in their applications.
